### PR TITLE
[WFLY-15343] More efficient/cleaner use of Maps

### PR DESCRIPTION
**Pull request by zodac-personal**
https://issues.redhat.com/browse/WFLY-15343

PR for the 'webservices' module

### CODE REVIEW

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_JMS.java [hunk 1]
```diff
@@ -92,37 +92,40 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
...
-                        QName serviceName = new QName(tns, service);
-                        QName portName = new QName(tns, port);
-                        String soapAddress = parser.filterSoapAddress(serviceName, portName, SOAPAddressWSDLParser.SOAP_OVER_JMS_NS);
-                        if (soapAddress != null) {
-                            ClassInfo webServiceClassInfo = (ClassInfo) ai.target();
-                            String beanClassName = webServiceClassInfo.name().toString();
-                            //service name ?
-                            list.add(new JMSEndpointMetaData(beanClassName, port, beanClassName, wsdlLocation, soapAddress));
-                        }
+        List<JMSEndpointMetaData> list = new LinkedList<>();
```
Comment (boris-unckel) [inline webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_JMS.java:95]: May be you can safe some resource here:
L94:  `List<JMSEndpointMetaData> list = Collections.emptyList();`
L95: keep the guard: `if (!map.isEmpty()) {`
L96: `list = new LinkedList<>();`

Comment (zodac) [reply webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_JMS.java:95]: I was thinking the same (existing early with an empty list if the map is empty), but the list gets passed into the constructor of `JMSEndpointsMetaData`, and that will replace any empty list with `Collections.emptyList()`. So I assume any unnecessary list will be garbage-collected away pretty quickly.

But I'm happy to change it if you feel it's worthwhile (saves other readers from needing to check how the list is used I guess).

Comment (boris-unckel) [reply webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_JMS.java:95]: @zodac The previous implementation had the guard and the receiver of the list `JMSEndpointsMetaData` checks for null and emptiness. I assume this is intentionally and avoiding unnecessary object creation/processing seems useful here. `Collections.emptyList()` returns a static object - no (unnecessary!) garbage-collection will happen.

Comment (zodac) [reply webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_JMS.java]: Sounds fair. I'll extract the loop logic to another method to make it clearer to return early with an empty list instead.

**Comment by darranl** [comment]
@zodac Can you please confirm if you are still working on this PR?

As per the comment from Brian this will need squashing back to a single commit as the change does not justify multiple commits, also we are still missing the confirmation of the MIT licence terms.

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_JMS.java [hunk 2]
```diff
@@ -82,48 +83,56 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
...
-                        QName serviceName = new QName(tns, service);
-                        QName portName = new QName(tns, port);
-                        String soapAddress = parser.filterSoapAddress(serviceName, portName, SOAPAddressWSDLParser.SOAP_OVER_JMS_NS);
-                        if (soapAddress != null) {
-                            ClassInfo webServiceClassInfo = (ClassInfo) ai.target();
-                            String beanClassName = webServiceClassInfo.name().toString();
-                            //service name ?
-                            list.add(new JMSEndpointMetaData(beanClassName, port, beanClassName, wsdlLocation, soapAddress));
-                        }
+        final List<JMSEndpointMetaData> list = extractSOAPOverJMSBindings(unit, map);
```
Comment (darranl) [inline webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_JMS.java:91]: Maybe it should have always been a read only list but at this point the list returned could now be read only OR modifiable.

**Comment by darranl** [comment]
I am going to close this PR as it has been dormant for a few months now, if you would like to continue please reopen.

**Pull request closed without merging**